native_client/dotnet/DeepSpeechClient/DeepSpeech.cs

```diff
@@ -61,16 +62,51 @@ public DeepSpeech()
             {
                 throw new FileNotFoundException(exceptionMessage);
             }
-            int result = NativeImp.DS_CreateModel(aModelPath,
+            var resultCode = NativeImp.DS_CreateModel(aModelPath,
                             aNCep,
                             aNContext,
                             aAlphabetConfigPath,
                             aBeamWidth,
                             ref _modelStatePP);
+            EvaluateResultCode(resultCode);
             _modelStateP = *_modelStatePP;
-            return result;
-
+        }
 
+        /// <summary>
+        /// Evaluate the result code and will raise an exception if necessary.
+        /// </summary>
+        /// <param name="resultCode">Native result code.</param>
+        private void EvaluateResultCode(ErrorCodes resultCode)
+        {
+            switch (resultCode)
+            {
+                case ErrorCodes.DS_ERR_OK:
+                    break;
+                case ErrorCodes.DS_ERR_NO_MODEL:
+                    throw new ArgumentException("Missing model information.");
+                case ErrorCodes.DS_ERR_INVALID_ALPHABET:
+                    throw new ArgumentException("Invalid alphabet file or invalid alphabet size.");
+                case ErrorCodes.DS_ERR_INVALID_SHAPE:
+                    throw new ArgumentException("Invalid model shape.");
+                case ErrorCodes.DS_ERR_INVALID_LM:
+                    throw new ArgumentException("Invalid language model file.");
```

Comment:
You also miss `DS_ERR_MODEL_INCOMPATIBLE` here

Reply:
Yes, was about to fix that